recording.js

```diff
@@ -57,82 +64,68 @@ function generateFrame(width, height, padding) {
-      .orama-window {
-        position: absolute;
-        margin: 0 auto;
-        z-index: 0;
-        display: block;
-        height: ${height + 40}px;
-        width: ${width}px;
-        bottom: 0;
-        right: 0;
-        left: 0;
-        top: calc(50% - 20px);
-        overflow: hidden;
-        transform: translateY(-50%);
-        background-color: ${backgroundColor};
-        color: ${foregroundColor};
-        -webkit-box-shadow: 0px 5px 41px -4px rgba(0,0,0,0.75);
-        -moz-box-shadow: 0px 5px 41px -4px rgba(0,0,0,0.75);
-        box-shadow: 0px 5px 41px -4px rgba(0,0,0,0.75);
-        border-radius: 8px;
-      }
-      .orama-title {
-        position: absolute;
-        display: block;
-        top: 0;
-        left: 0;
-        width: 100%;
-        height: 40px;
-        background-color: ${backgroundColor};
-        padding: 10px;
-      }
-      .orama-dot {
-        position: absolute;
-        left: 0;
-        display: inline-block;
-        width: 20px;
-        height: 20px;
-        border-radius: 100%;
-        background-color: white;
-      }
-      .orama-close {
-        left: 10px;
-        background-color: #ff704c;
-      }
-      .orama-max {
-        left: 40px;
-        background-color: #fff94c;
-      }
-      .orama-min {
-        left: 70px;
-        background-color: #4dff35;
+      :root {
+          --orama-size: ${frameSize}px;
+          --orama-button-size: calc(var(--orama-size) / 3.5);
+          --orama-title-size: var(--orama-size);
+          --orama-window-height: ${height}px;
+          --orama-window-width: ${width}px;
+          --orama-title-radius: ${frameRadius}px;
+          --orama-window-color: ${backgroundColor};
+          --orama-close-color: rgb(255, 95, 86);
+          --orama-max-color: rgb(255, 189, 46);
+          --orama-min-color: rgb(39, 201, 63);
       }
     </style>
-    <div class="orama-window">
-      <div class="orama-title">
-        <div class="orama-dot orama-close"></div>
-        <div class="orama-dot orama-max"></div>
-        <div class="orama-dot orama-min"></div>
-      </div>
-    </div>
-  `;
+    ${frameMarkup}`;
   document.body.appendChild(frame);
 
-  /* Paint baseCanvas */
+  // Create Canvas for drawing
+  const baseCanvas = document.createElement('canvas');
+  baseCanvas.height = height + frameSize + padding * 2;
+  baseCanvas.width = width + padding * 2;
+
+  /* Create shadow behind terminal window */
+  const baseCtx = baseCanvas.getContext('2d');
+  baseCtx.fillStyle = 'white';
+  baseCtx.fillRect(0, 0, baseCanvas.width, baseCanvas.height);
+  baseCtx.shadowColor = 'rgba(0, 0, 0, 0.25)';
+  baseCtx.shadowBlur = width / 10;
+  baseCtx.fillStyle = backgroundColor;
+  baseCtx.fillRect(padding, padding + frameSize, width, height - frameRadius);
+
+  /* Paint Frame onto Canvas */
   return html2canvas(document.getElementById('orama-frame'), {
-    canvas: baseCanvas,
-    height: height + padding * 2,
+    backgroundColor: null,

```

Comment:
Why is setting backgroundColor to null needed?

Reply:
This is done for overlaying the HTML elements with a transparent background as seen in the [html2canvas documentation](https://html2canvas.hertzen.com/configuration). I agree that it's a poor API choice.